src/Nethermind/Nethermind.Evm/Tracing/GasEstimator.cs

```diff
@@ -35,12 +33,7 @@ public long Estimate(Transaction tx, BlockHeader header, EstimateGasTracer gasTr
         {
             IReleaseSpec releaseSpec = _specProvider.GetSpec(header.Number + 1, header.Timestamp + _blocksConfig.SecondsPerSlot);
 
-            long intrinsicGas = tx.GasLimit - gasTracer.IntrinsicGasAt;
-            if (tx.GasLimit > header.GasLimit)
-            {
-                return Math.Max(intrinsicGas, gasTracer.GasSpent + gasTracer.CalculateAdditionalGasRequired(tx, releaseSpec));
-            }
-
```

Comment:
This wasn't bad edge case fallback if we didn't have that gas limit discrepancy, maybe worth to keep it?